File: algorithms/python/complementofBase10Integer/complementofBase10Integer.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/complement-of-base-10-integer/
# Author: Miao Zhang
# Date:   2021-04-01

class Solution:
    def bitwiseComplement(self, N: int) -> int:
        num = 1
        while num < N:
            num = (num << 1) + 1
        return num ^ N

File: algorithms/cpp/complementofBase10Integer/complementofBase10Integer.cpp
// Source: https://leetcode.com/problems/complement-of-base-10-integer/
// Author: Miao Zhang
// Date:   2021-04-01

class Solution {
public:
    int bitwiseComplement(int N) {
        int num = 1;
        while (num < N) {
            num = (num << 1) + 1;
        }
        // num = 2 ** n - 1
        return num ^ N;
    }
};
